crypto/ct-aead/src/aes_impl.rs

```diff
@@ -0,0 +1,41 @@
+use crate::CtAeadDecrypt;
+
+use aes_gcm::{AesGcm, Tag, A_MAX, C_MAX};
+use block_cipher::{
+    consts::{U0, U16},
+    generic_array::{ArrayLength, GenericArray},
+    Block, BlockCipher,
+};
+
+impl<Aes, NonceSize> CtAeadDecrypt for AesGcm<Aes, NonceSize>
+where
+    Aes: BlockCipher<BlockSize = U16>,
+    Aes::ParBlocks: ArrayLength<Block<Aes>>,
+    NonceSize: ArrayLength<u8>,
+{
+    type NonceSize = NonceSize;
+    type TagSize = U16;
+    type CiphertextOverhead = U0;
+
+    fn ct_decrypt_in_place_detached(
```

Comment:
Might be good to reference the original decrypt_in_place_detached implementation,
https://docs.rs/aes-gcm/0.6.0/src/aes_gcm/lib.rs.html#251 . Reading the original helped me see what's changed here.

Reply:
Done.